apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -322,10 +377,46 @@ def mlset(self, keys, value):
         container, key = find_ml(self._data, keys)
         container[key] = value
 
+    def mlset_cast(self, keys, value):
+        val_type = self.find_type(keys)
+        if val_type:
+            if val_type == "integer":
+                try:
+                    value = int(value)
+                except ValueError as err:
+                    err.value_type = val_type
+                    raise err
+            elif val_type == "boolean":
+                try:
+                    value = to_bool(value)
+                except ValueError as err:
```

Comment:
If you move this between `if val_type:` and `if val_type == "integer":`, then you would only need one try..except.